forem/forem
# [EOY 2020 Issue] Alert admins in modal when enabling Auth Provider without keys

## What type of PR is this? (check all applicable)


## Description
This PR alerts admins via a modal when they attempt to enable Auth Providers without entering their respective key and secret. The PR also refactors the `adminModal.js` function to make it more generalizable for more use-cases.

## Related Tickets & Documents
This PR introduces an improved User Flow, that will replace the Warning Banner in [this PR](https://github.com/forem/forem/pull/11696).

## QA Instructions, Screenshots, Recordings
1. As `:super_admin` role, navigate to `localhost:xxxx/admin/config`. Toggle open the Authentication section.
2. Enable one or more Authentication Providers, but leave the keys blank (or you can type random string for Key OR Secret, but NOT both). Update Site Configuration. A modal should popup listing the providers you attempted to enable.
3. Click "Continue editing" button or "X" the modal. You should be taken back to the main Config page AND the Update Site Configuration button should NOT be disabled.
4. Update site configuration again. This time, click "Save anyway". The update action should fire, the page refresh, but the keyless provider(s) you attempted to enable should still be disabled.
5. Re-enable your chosen providers, but this time, enter dummy strings in both key fields. Update site configuration. The page should refresh, and you should see your chosen providers are enabled.

Demo Video of PR:
https://www.loom.com/share/2ffd7af973af4054b21832f59cb9eec9

### UI accessibility concerns?
None anticipated

## Added tests?

- [X] No, and this is why: _current tests cover the changes in this PR_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [X] No documentation needed; docs already reflect the need to enter auth provider key and secret

## [optional] Are there any post deployment tasks we need to perform?
No

## [optional] What gif best describes this PR or how it makes you feel?

![Toddler pushing object](https://media.giphy.com/media/YPKFBSrq0EoqPJGqTn/giphy-downsized.gif)
